chatterbot/trainers.py

```diff
@@ -78,17 +77,25 @@ def train(self, conversation):
         Train the chat bot based on the provided list of
         statements that represents a single conversation.
         """
-        previous_statement_text = None
 
-        for text in conversation:
-            statement = self.get_or_create(text)
+        # Inline method yields statements for larger datasets
+        def get_statements(conversations):
+            previous_statement_text = None
+
+            for text in conversation:
+                statement = self.get_or_create(text)
+
+                if previous_statement_text:
+                    statement.add_response(
+                        Response(previous_statement_text)
+                    )
+
+                previous_statement_text = statement.text
+                yield statement
```

Comment:
> Any more information about lazy lists

This response on Stack Overflow seems to sum up generators fairly well https://stackoverflow.com/questions/1756096/understanding-generators-in-python